schemas/pubsub/schema.json

```diff
@@ -0,0 +1,105 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://raw.githubusercontent.com/google/knative-gcp/master/schemas/pubsub/schema.json",
+  "title": "The PubSub Schema",
+  "definitions": {
+    "push": {
+      "$id": "#push",
+      "type": "object",
+      "required": [
+        "subscription",
+        "message"
+      ],
+      "properties": {
+        "subscription": {
+          "type": "string",
+          "title": "The Subscription Schema",
+          "default": "",
+          "examples": [
+            "cre-pull-176e54ba-bf7b-11e9-ae05-42010a8a00e1"
+          ],
+          "pattern": "^(.*)$"
+        },
+        "message": {
+          "type": "object",
+          "title": "The Message Schema",
+          "required": [
+            "id",
+            "data",
+            "publish_time"
+          ],
+          "properties": {
+            "id": {
+              "type": "string",
+              "title": "The Id Schema",
+              "default": "",
+              "examples": [
+                "684198665501459"
+              ],
+              "pattern": "^(.*)$"
+            },
+            "data": {
+              "title": "The Data Schema",
+              "oneOf": [
+                {
+                  "type": "string",
+                  "default": "",
+                  "examples": [
+                    "hello world",
+                    "aGVsbG8gd29ybGQ="
+                  ],
+                  "pattern": "^(.*)$"
+                },
+                {
+                  "type": "object"
+                }
+              ]
+            },
+            "publish_time": {
+              "type": "string",
+              "title": "The Publish_time Schema",
+              "default": "",
+              "examples": [
+                "2019-08-15T16:40:09.932Z"
+              ],
+              "pattern": "^(.*)$"
+            }
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "binary": {
+      "$id": "#binary",
+      "oneOf": [
```

Comment:
binary and structured are the same for data. But data can be many more things and encoded in many more ways. Are we talking about the http version of the data? or the data converted?

Reply:
Agreed to remove the schema, and promote the schema if provided in the event...
We'll set the mode as an extension attribute as well